src/archive.cpp

```diff
@@ -86,6 +86,45 @@ namespace zim
     return ret;
   }
 
+  Item Archive::getIllustrationItem(unsigned int width, unsigned int height) const {
+    std::ostringstream ss;
+    ss  << "Illustration_" << width << "x" << height;
+    auto r = m_impl->findx('M', ss.str());
+    if (r.first) {
+      return getEntryByPath(entry_index_type(r.second)).getItem();
+    }
+    // We haven't found the exact entry. Let's "search" for a illustration and
+    // use the first one we found.
+#if 0
+    // We have decided to not implement fallback in case of wrong resolution for now.
+    // We keep this code for reference.
+    r = m_impl->findx('M', "Illustration");
+    auto entry = getEntryByPath(entry_index_type(r.second));
+    if (entry.getPath().find("Illustration") == 0) {
+      return entry.getItem();
+    }
+#endif
+    // Check know favicon for older zims.
+    for(auto ns:{'-', 'I'}) {
+      for (auto& path:{"favicon", "favicon.png"}) {
+        auto r = m_impl->findx(ns, path);
+        if (r.first) {
+          return getEntryByPath(entry_index_type(r.second)).getItem(true);
+        }
+      }
+    }
```

Comment:
So `getIllustrationItem(width, height)` can silently return an item of a different size. Maybe the return type must be adjusted so that the caller knows what (s)he got? Of course that can be guessed from the item's path, but having it in the interface is more fool-proof.